python_utils.py

```diff
@@ -374,6 +377,54 @@ def convert_to_bytes(string_to_convert):
     Returns:
         bytes. The encoded string.
     """
-    if isinstance(string_to_convert, STR):
+    if isinstance(string_to_convert, UNICODE):
         return string_to_convert.encode('utf-8')
     return bytes(string_to_convert)
+
+
+def _recursively_convert_to_str(value):
+    """Convert all builtins.bytes and builtins.str elements in a data structure
+    to bytes and unicode respectively. This is required for the
+    yaml.safe_dump() function to work as it only works for unicode and bytes and
+    not builtins.bytes nor builtins.str(UNICODE). See:
+    https://stackoverflow.com/a/1950399/11755830
+
+    Args:
+        value: list|dict|BASESTRING. The data structure to convert.
+
+    Returns:
+        list|dict|bytes|unicode. The data structure in bytes and unicode.
+    """
+    if isinstance(value, list):
+        return [_recursively_convert_to_str(e) for e in value]
+    elif isinstance(value, dict):
+        return {
+            _recursively_convert_to_str(k): _recursively_convert_to_str(
+                v) for k, v in value.items()}
+    # We are using 'type' here instead of 'isinstance' because we need to
+    # clearly distinguish the builtins.str and builtins.bytes strings.
+    elif type(value) == future.types.newstr:  # pylint: disable=unidiomatic-typecheck
+        temp = str(value.encode('utf-8'))
+        # Remove the b'' prefix from the string.
+        return temp[2:-1].decode('utf-8')
+    elif type(value) == future.types.newbytes:  # pylint: disable=unidiomatic-typecheck
+        temp = bytes(value)
+        # Remove the b'' prefix from the string.
+        return temp[2:-1]
+    else:
+        return value
+
+
+def yaml_from_dict(dictionary, width=80):
```

Comment:
I think it was fine to keep this in `utils` since the Python2-3 specific logic was placed in `_recursively_convert_to_str`.

Reply:
@seanlip said to move this to python_utils since no other function should call _recursively_convert_to_str

Reply:
OK, sgtm.